src/omefiles.cpp

```diff
@@ -26,67 +26,139 @@
+      }, "Get the size of the X dimension.")
+    .def("get_size_y", [](OMETIFFReader &r) {
+        return r.getSizeY();
+      }, "Get the size of the Y dimension.")
+    .def("get_size_z", [](OMETIFFReader &r) {
+        return r.getSizeZ();
+      }, "Get the size of the Z dimension.")
+    .def("get_size_t", [](OMETIFFReader &r) {
+        return r.getSizeT();
+      }, "Get the size of the T dimension.")
+    .def("get_size_c", [](OMETIFFReader &r) {
+        return r.getSizeC();
+      }, "Get the size of the C dimension.")
+    .def("get_dimension_order", [](OMETIFFReader &r) {
+        return r.getDimensionOrder();
+      }, "Get a five-character string representing the order in which "
+      "planes will be returned.")
+    .def("get_index", [](OMETIFFReader &r, size_t z, size_t c, size_t t) {
+	return r.getIndex(z, c, t);
+      }, "Get the plane index corresponding to the given ZCT coordinates.")
+    .def("get_zct_coords", [](OMETIFFReader &r, size_t index) {
+	return r.getZCTCoords(index);
+      }, "Get the ZCT coordinates for the given plane index.")
+    .def("get_rgb_channel_count", [](OMETIFFReader &r, size_t channel) {
+	return r.getRGBChannelCount(channel);
+      }, "Get the number of sub-channels for the given channel.")
+    .def("is_interleaved",
+	 (bool (OMETIFFReader::*)(size_t) const) &OMETIFFReader::isInterleaved,
+	 "Whether or not the given channel is interleaved")
+    .def("is_interleaved",
+	 (bool (OMETIFFReader::*)() const) &OMETIFFReader::isInterleaved,
+	 "Whether or not the channels are interleaved")
+    .def("get_pixel_type", [](OMETIFFReader &r) {
+	switch(r.getPixelType()) {
+	case ome::xml::model::enums::PixelType::INT8:
+	  return "i1";
+	case ome::xml::model::enums::PixelType::INT16:
+	  return "i2";
+	case ome::xml::model::enums::PixelType::INT32:
+	  return "i4";
+	case ome::xml::model::enums::PixelType::UINT8:
+	  return "u1";
+	case ome::xml::model::enums::PixelType::UINT16:
+	  return "u2";
+	case ome::xml::model::enums::PixelType::UINT32:
+	  return "u4";
+	case ome::xml::model::enums::PixelType::FLOAT:
+	  return "f4";
+	case ome::xml::model::enums::PixelType::DOUBLE:
+	  return "f8";
+	case ome::xml::model::enums::PixelType::COMPLEXFLOAT:
+	  return "c8";
+	case ome::xml::model::enums::PixelType::COMPLEXDOUBLE:
+	  return "c16";
+	case ome::xml::model::enums::PixelType::BIT:
+	  return "b1";
+	default:
+	  throw std::invalid_argument("unknown pixel type");
+	}
+      }, "Get the pixel type.")
+    .def("open_bytes", [](OMETIFFReader &r, size_t plane) {
+	ome::files::VariantPixelBuffer buf;
+	r.openBytes(plane, buf);
+	return py::bytes(std::string(
+          reinterpret_cast<char*>(buf.data()),
+          buf.num_elements() * ome::files::bytesPerPixel(buf.pixelType())
+	));
+      }, "Obtain the image plane for the given index.")
+    .def("get_used_files", [](OMETIFFReader &r, bool noPixels = false) {
+	std::vector<std::string> fnames;
+	for (const auto &f : r.getUsedFiles(noPixels)) {
```

Comment:
Discussed in person, but mentioning here for anyone interested.  An alternative here would be to use `std::transform` with a `back_inserter` and operate on the whole container without looping (visibly).  But it adds little additional clarity compared with a range-based for loop, and so what is here is fine.